src/vm/eepolicy.cpp

```diff
@@ -1176,18 +1176,31 @@ inline void LogCallstackForLogWorker()
 // Return Value:
 //    None
 //
-inline void DoLogForFailFastException(LPCWSTR pszMessage, PEXCEPTION_POINTERS pExceptionInfo)
+inline void DoLogForFailFastException(LPCWSTR pszMessage, PEXCEPTION_POINTERS pExceptionInfo, UINT errorSource)
 {
     WRAPPER_NO_CONTRACT;
 
     Thread *pThread = GetThread();
     EX_TRY
     {
-        PrintToStdErrA("FailFast: ");
+        if (errorSource == EEPolicy::FFES_FailFast)
+        {
+            PrintToStdErrA("FailFast: ");
+        }
+        else if (errorSource == EEPolicy::FFES_FailFast)
+        {
+            PrintToStdErrA("Assertion Failed:");
+        }
+        else if (errorSource == EEPolicy::FFES_FailFast)
+        {
+            PrintToStdErrA("Assumption Failed:");
+        }
+
+        PrintToStdErrA("\n");
         PrintToStdErrW((WCHAR*)pszMessage);
         PrintToStdErrA("\n");
 
-        if (pThread)
+        if (pThread && errorSource == EEPolicy::FFES_FailFast)
```

Comment:
This looks like that we will print stack trace for assertion failed, and then the text of the error. It would be nice always have the error first - it is how it always worked.